cdap-api/src/main/java/co/cask/cdap/api/metadata/MetadataReaderContext.java

```diff
@@ -1,69 +0,0 @@
-/*
- * Copyright © 2018 Cask Data, Inc.
- *
- * Licensed under the Apache License, Version 2.0 (the "License"); you may not
- * use this file except in compliance with the License. You may obtain a copy of
- * the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
- * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
- * License for the specific language governing permissions and limitations under
- * the License.
- */
-package co.cask.cdap.api.metadata;
-
-import java.util.Map;
-import java.util.Set;
-import javax.ws.rs.NotFoundException;
-
-/**
- * The context for reading metadata from program
- */
-public interface MetadataReaderContext {
```

Comment:
By removing this entirely, that means there no API to read meta data? I think we need a way to allow:
- in prepareLaunch, read and write metadata and mark it for use at runtime 
- at runtime (including initialize/destroy), read metadata marked in prepareLaunch, submit writes for meta data
- in postExecution, read and write metadata (and that may include meta data changed at runtime)